test/cache.js

```diff
@@ -0,0 +1,58 @@
+var test = require("tape")
+var webpack = require("webpack")
+var assign = require("object-assign")
+var conf = require("./utils/conf")
+var fs = require("fs")
+
+var cacheFilePath = "./node_modules/.cache/eslint-loader/data.json"
+
+test("eslint-loader can cache results", function(t) {
+
+  // delete the require cache for eslint-loader otherwise any previously run
+  // tests will have initialised the cache as false and prevent this test
+  // from creating the cache file
+  delete require.cache[require.resolve("../index.js")]
```

Comment:
That sounds sensible but out of scope for this PR. I'll create a new issue to gather thoughts on a restructure to testing and then this PR can wait for that.